crates/next-core/src/next_config.rs

```diff
@@ -29,29 +31,212 @@ use turbopack_node::{
 
 use crate::embed_js::next_asset;
 
-#[turbo_tasks::value(serialization = "custom")]
-#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+#[turbo_tasks::value(serialization = "custom", eq = "manual")]
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct NextConfig {
-    pub config_file: Option<String>,
+    pub cross_origin: Option<String>,
+    pub config_file: String,
     pub config_file_name: String,
-    pub typescript: Option<TypeScriptConfig>,
+
     pub react_strict_mode: Option<bool>,
-    pub experimental: Option<ExperimentalConfig>,
-    pub env: Option<HashMap<String, String>>,
-    pub compiler: Option<CompilerConfig>,
+    pub experimental: ExperimentalConfig,
     pub images: ImageConfig,
+
+    #[serde(flatten)]
+    unsupported: UnsupportedNextConfig,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(rename_all = "camelCase")]
+struct AmpConfig {
+    canonical_base: Option<String>,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(rename_all = "camelCase")]
+struct EslintConfig {
+    dirs: Option<Vec<String>>,
+    ignore_during_builds: Option<bool>,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(rename_all = "kebab-case")]
+enum BuildActivityPositions {
+    BottomRight,
+    BottomLeft,
+    TopRight,
+    TopLeft,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(rename_all = "camelCase")]
+struct DevIndicatorsConfig {
+    build_activity: bool,
+    build_activity_position: BuildActivityPositions,
 }
 
-#[derive(Clone, Debug, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs)]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(rename_all = "camelCase")]
+struct OnDemandEntriesConfig {
+    max_inactive_age: f64,
```

Comment:
Are these floats or u64s?

Reply:
yeah, ideally these would be `u64`, but for JS compat I've put them in here as `f64` for now